test/k8s-integration/main.go

```diff
@@ -216,55 +241,91 @@ func clusterUp(k8sDir string) error {
+	}
 
-	out, err := exec.Command("cp", controllerFile, filepath.Join(kubeManifestDir, "controller_ss.yaml")).CombinedOutput()
+	// Install kustomize
+	out, err = exec.Command(filepath.Join(pkgDir, "deploy", "kubernetes", "install-kustomize.sh")).CombinedOutput()
 	if err != nil {
-		return fmt.Errorf("failed to copy controller manifest: %s, err: %v", out, err)
+		return fmt.Errorf("failed to install kustomize: %s, err: %v", out, err)
 	}
 
-	out, err = exec.Command("cp", nodeFile, filepath.Join(kubeManifestDir, "node_ds.yaml")).CombinedOutput()
+	// Edit ci kustomization to use given image tag
+	overlayDir := getOverlayDir(pkgDir, deployOverlayName)
+	err = os.Chdir(overlayDir)
 	if err != nil {
-		return fmt.Errorf("failed to copy node manifest: %s, err: %v", out, err)
+		return fmt.Errorf("failed to change to overlay directory: %s, err: %v", out, err)
 	}
 
-	// Change manifest container to staging image and version
-	err = mutateManifests(filepath.Join(kubeManifestDir, "node_ds.yaml"), filepath.Join(kubeManifestDir, "controller_ss.yaml"), stagingImage, stagingVersion)
+	// TODO: in a local environment this is going to modify the actual kustomize files.
+	// maybe a copy should be made instead
+	out, err = exec.Command(
+		filepath.Join(pkgDir, "bin", "kustomize"),
+		"edit",
+		"set",
+		"imagetag",
+		fmt.Sprintf("%s:%s", stagingImage, stagingVersion)).CombinedOutput()
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to edit kustomize: %s, err: %v", out, err)
 	}
-	return nil
-}
 
-func mutateManifests(nodeFile, controllerFile, stagingImage, stagingVersion string) error {
-	// TODO: Make this existing image configurable, ideally we aren't actually reliant on
-	// a hardcoded image to replace.
+	// setup service account file for secret creation
+	tmpSaFile := fmt.Sprintf("/tmp/%s/cloud-sa.json", string(uuid.NewUUID()))
 
-	// The actual solution we need to look into doing this is using Kustomize
-	// to mutate the YAML files
-	existingImage := "gcr.io/dyzz-csi-staging/csi/gce-pd-driver:latest"
+	os.MkdirAll(filepath.Dir(tmpSaFile), 0750)
+	defer os.Remove(filepath.Dir(tmpSaFile))
 
-	escapedImage := strings.Replace(stagingImage, "/", "\\/", -1)
-	escapedVersion := strings.Replace(stagingVersion, "/", "\\/", -1)
-	escapedExistingImage := strings.Replace(existingImage, "/", "\\/", -1)
+	// Need to copy it to name the file "cloud-sa.json"
+	out, err = exec.Command("cp", *saFile, tmpSaFile).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error copying service account key: %s, err: %v", out, err)
+	}
+	defer shredFile(tmpSaFile)
+
+	// deploy driver
+	deployCmd := exec.Command(filepath.Join(pkgDir, "deploy", "kubernetes", "deploy-driver.sh"), "--skip-sa-check")
+	deployCmd.Env = append(os.Environ(),
+		fmt.Sprintf("GOPATH=%s", goPath),
+		fmt.Sprintf("GCE_PD_SA_DIR=%s", filepath.Dir(tmpSaFile)),
+		fmt.Sprintf("GCE_PD_DRIVER_VERSION=%s", deployOverlayName),
+	)
+	out, err = deployCmd.CombinedOutput()
```

Comment:
consider using `runLongRunningCommand()`, and maybe changing the name of that function as well. It's not really just for long running commands it just redirects stdout/stdin for the script to the os.

Reply:
done